tasks/deploy/lib/environment-extensions.ts

```diff
@@ -28,75 +28,72 @@ export type HyperdriveDeployRuntimeOptions = {
 };
 
 /**
- * Type representing the Hardhat Task params accepted by all Hyperdrive deploy tasks
+ * Represents the parameters used by all Hyperdrive deploy tasks
  */
-export type HyperdriveDeployBaseTaskParams = HyperdriveDeployRuntimeOptions & {
+export type HyperdriveDeployBaseTaskParams = HyperdriveDeployRuntimeOptions;
+
+/**
+ * Represents the parameters for a Hyperdrive deploy task requiring a name
+ */
+export type HyperdriveDeployNamedTaskParams = HyperdriveDeployRuntimeOptions & {
     // Name of the primary contract to deploy
     //  - supporting contract names will be derived from it if applicable
     name: string;
 };
 
 /**
- * Base Hyperdrive deploy task with all base params already added.
+ * Base Hyperdrive deploy task with all base params.
  */
 export const HyperdriveDeployBaseTask = (task: ConfigurableTaskDefinition) =>
-    task
-        .addParam(
-            "name",
-            "name of the primary contract to deploy (supporting contract names will be derived from it)",
-            undefined,
-            types.string,
-        )
-        .addOptionalParam(
-            "noSave",
-            "skip saving deployment artifacts and data",
-            false,
-            types.boolean,
-        )
-        .addOptionalParam(
-            "overwrite",
-            "overwrite deployment artifacts if they exist",
-            false,
-            types.boolean,
-        );
-
-export type HyperdriveDeployCoordinatorRuntimeOptions =
-    HyperdriveDeployRuntimeOptions & {
-        approveWithFactory?: boolean;
-    };
-
-export const HyperdriveDeployCoordinatorTask = (
-    task: ConfigurableTaskDefinition,
-) =>
-    HyperdriveDeployBaseTask(task).addOptionalParam(
-        "approveWithFactory",
-        "attempt approving the HyperdriveDeployerCoordinator with its factory (will fail if deployer address is not governance)",
+    task.addOptionalParam(
```

Comment:
When do we not want to save? This is a genuine question, I'm just curious

Reply:
Original idea was for local or hardhat network deploys so the `deployments.local.json` file wouldn't have to be deleted between subsequent runs.... however the deployment process is so dependent on checking previous deployment entries that this strategy wouldn't work. Will remove